### PR TITLE
Add positioner to center on a selected annotation

### PR DESCRIPTION
**Pull request by ronnyroeller**
When working with annotations (extention-annotation), it's handy to have a positioner that places a popup in the middle of the selected annotation. This allows e.g. to add a popup to remove/change the annotation when the user clicks on it.

@see https://discord.com/channels/726035064831344711/745695557976195072/745697774355480606

### Checklist

- [x] I have read the [**contributing**](https://github.com/remirror/remirror/blob/HEAD/docs/contributing.md) document.
- [x] My code follows the code style of this project and `pnpm fix` completed successfully.
- [x] I have updated the documentation where necessary.
- [x] ~New code is unit tested~ and all current tests pass when running `pnpm test`. - [Discussed with @ifiokjr](https://discord.com/channels/726035064831344711/745695557976195072/745762246013091922) to not test the code for now

### CODE REVIEW

File: packages/@remirror/extension-annotation/src/positioners.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import {
+  Coords,
+  emptyVirtualPosition,
+  hasStateChanged,
+  Positioner,
+} from 'remirror/extension/positioner';
```
Comment (ifiokjr) [inline packages/@remirror/extension-annotation/src/positioners.ts:6]: Internally you should use the `@remirror/extension-positioner`.

```suggestion
} from '@remirror/extension-positioner';
```

File: packages/@remirror/extension-annotation/package.json [hunk 2]
```diff
@@ -28,10 +28,12 @@
   },
   "devDependencies": {
     "@remirror/core": "1.0.0-next.22",
+    "@remirror/extension-positioner": "*",
```
Comment (ifiokjr) [inline packages/@remirror/extension-annotation/package.json:31]: I think the linting is failing due to this. I'm not sure why `pnpm fix` sometimes doesn't fix things properly.

```suggestion
    "@remirror/extension-positioner": "1.0.0-next.22",
```

**Review by ifiokjr — APPROVED** [review]

File: packages/@remirror/extension-annotation/package.json (hunk 2 above)
Comment (ifiokjr) [reply packages/@remirror/extension-annotation/package.json]: My bad, I forgot that it wouldn't automatically update the lock file. This is a problem with auto suggestions. They seldom work without further adjustments 😄

**Comment by ronnyroeller** [comment]
@ifiokjr Thanks for the review. You assigned the ticket to me. Is there something specific you want me to do before this PR can get merged?

**Comment by ifiokjr** [comment]
Oh, I assigned it to you since you're working on it to be honest. 

Although the build is currently failing because the `pnpm-lock.yaml` file needs to be updated.

**Comment by ronnyroeller** [comment]
> Oh, I assigned it to you since you're working on it, no special reason. It' helps with tracking on the [project board](https://github.com/orgs/remirror/projects/1#card-43991555).
> 
> Although the build is currently failing because the `pnpm-lock.yaml` file needs to be updated.

Sorry, I had missed that one. Fixed.

**Pull request merged**